crates/admin/src/rest_api/deployments.rs

```diff
@@ -45,72 +47,126 @@ use serde::Deserialize;
     )
 )]
-pub async fn create_deployment(
-    State(state): State<AdminServiceState>,
+pub async fn create_deployment<V>(
+    State(state): State<AdminServiceState<V>>,
     #[request_body(required = true)] Json(payload): Json<RegisterDeploymentRequest>,
 ) -> Result<impl IntoResponse, MetaApiError> {
-    let (discover_endpoint, force, dry_run) = match payload {
-        RegisterDeploymentRequest::Http {
-            uri,
-            additional_headers,
-            force,
-            dry_run,
-        } => (
-            DiscoverEndpoint::new(
-                Endpoint::Http(uri, Default::default()),
-                additional_headers.unwrap_or_default().into(),
-            ),
-            force,
-            dry_run,
-        ),
-        RegisterDeploymentRequest::Lambda {
-            arn,
-            assume_role_arn,
-            additional_headers,
-            force,
-            dry_run,
-        } => (
-            DiscoverEndpoint::new(
-                Endpoint::Lambda(
-                    arn.parse().map_err(|e: InvalidLambdaARN| {
-                        MetaApiError::InvalidField("arn", e.to_string())
-                    })?,
-                    assume_role_arn.map(Into::into),
+    state
+        .task_center
+        .run_in_scope("create-deployment", None, async {
+            let (discover_endpoint, force, dry_run) = match payload {
+                RegisterDeploymentRequest::Http {
+                    uri,
+                    additional_headers,
+                    force,
+                    dry_run,
+                } => (
+                    DiscoverEndpoint::new(
+                        Endpoint::Http(uri, Default::default()),
+                        additional_headers.unwrap_or_default().into(),
+                    ),
+                    force,
+                    dry_run,
                 ),
-                additional_headers.unwrap_or_default().into(),
-            ),
-            force,
-            dry_run,
-        ),
-    };
+                RegisterDeploymentRequest::Lambda {
+                    arn,
+                    assume_role_arn,
+                    additional_headers,
+                    force,
+                    dry_run,
+                } => (
+                    DiscoverEndpoint::new(
+                        Endpoint::Lambda(
+                            arn.parse().map_err(|e: InvalidLambdaARN| {
+                                MetaApiError::InvalidField("arn", e.to_string())
+                            })?,
+                            assume_role_arn.map(Into::into),
+                        ),
+                        additional_headers.unwrap_or_default().into(),
+                    ),
+                    force,
+                    dry_run,
+                ),
+            };
 
-    let apply_changes = if dry_run {
-        ApplyMode::DryRun
-    } else {
-        ApplyMode::Apply
-    };
+            let discovered_metadata = state
```

Comment:
let's add a comment with the implications of that, as discussed offline.

Reply:
Will do 👍